tests/components/deconz/test_climate.py

```diff
@@ -0,0 +1,140 @@
+
+
+SENSOR = {
+    "1": {
+        "id": "Climate 1 id",
+        "name": "Climate 1 name",
+        "type": "ZHAThermostat",
+        "state": {"on": True, "temperature": 2260},
+        "config": {"battery": 100, "heatsetpoint": 2200, "mode": "auto",
+                   "offset": 10, "valve": 30},
+        "uniqueid": "00:00:00:00:00:00:00:00-00"
+    },
+    "2": {
+        "id": "Sensor 2 id",
+        "name": "Sensor 2 name",
+        "type": "ZHAPresence",
+        "state": {"presence": False},
+        "config": {}
+    }
+}
+
+ENTRY_CONFIG = {
+    deconz.const.CONF_ALLOW_CLIP_SENSOR: True,
+    deconz.const.CONF_ALLOW_DECONZ_GROUPS: True,
+    deconz.config_flow.CONF_API_KEY: "ABCDEF",
+    deconz.config_flow.CONF_BRIDGEID: "0123456789",
+    deconz.config_flow.CONF_HOST: "1.2.3.4",
+    deconz.config_flow.CONF_PORT: 80
+}
+
+
+async def setup_gateway(hass, data, allow_clip_sensor=True):
+    """Load the deCONZ sensor platform."""
+    from pydeconz import DeconzSession
+    loop = Mock()
+    session = Mock()
+
+    ENTRY_CONFIG[deconz.const.CONF_ALLOW_CLIP_SENSOR] = allow_clip_sensor
+
+    config_entry = config_entries.ConfigEntry(
+        1, deconz.DOMAIN, 'Mock Title', ENTRY_CONFIG, 'test',
+        config_entries.CONN_CLASS_LOCAL_PUSH)
+    gateway = deconz.DeconzGateway(hass, config_entry)
+    gateway.api = DeconzSession(loop, session, **config_entry.data)
+    gateway.api.config = Mock()
+    hass.data[deconz.DOMAIN] = gateway
+
+    with patch('pydeconz.DeconzSession.async_get_state',
+               return_value=mock_coro(data)):
+        await gateway.api.async_load_parameters()
+
+    await hass.config_entries.async_forward_entry_setup(
+        config_entry, 'climate')
+    # To flush out the service call to update the group
+    await hass.async_block_till_done()
+
+
+async def test_platform_manually_configured(hass):
+    """Test that we do not discover anything or try to set up a gateway."""
+    assert await async_setup_component(hass, climate.DOMAIN, {
+        'climate': {
+            'platform': deconz.DOMAIN
+        }
+    }) is True
+    assert deconz.DOMAIN not in hass.data
+
+
+async def test_no_sensors(hass):
+    """Test that no sensors in deconz results in no climate entities."""
+    await setup_gateway(hass, {})
+    assert not hass.data[deconz.DOMAIN].deconz_ids
+    assert not hass.states.async_all()
+
+
+async def test_climate_devices(hass):
+    """Test successful creation of sensor entities."""
+    await setup_gateway(hass, {"sensors": SENSOR})
+    assert "climate.climate_1_name" in hass.data[deconz.DOMAIN].deconz_ids
+    assert "sensor.sensor_2_name" not in hass.data[deconz.DOMAIN].deconz_ids
+    assert len(hass.states.async_all()) == 1
+
+    hass.data[deconz.DOMAIN].api.sensors['1'].async_update(
+        {'state': {'on': False}})
+
+    await hass.services.async_call('climate', 'turn_on', {
+        'entity_id': 'climate.climate_1_name'
+    }, blocking=True)
+
```

Comment:
We should assert the state of the entity in the state machine between each service call.

Reply:
Any reference I can look at for this?

Reply:
Thinking about it. There is no setting of the state. All states are updated over websocket directly from Deconz

Reply:
How or when when will the state change after the service call be reflected in home assistant?

Reply:
When something happens in deconz and it signals over websocket

Reply:
We should test that the deconz callback and device state will represent the correct state in home assistant. It doesn't have to be in this test since we would mock the device state changes after the service call anyway, but we should test it somewhere.

Reply:
Since there's no direct feedback on state after the service call we should instead assert the calls on the mocked device after the service calls.

Reply:
Can I do that on a separate PR? That would require some work and it's better to contain this PR.

Reply:
I'm not sure what comment of mine you're referring to, but too both I'd answer:
We should test it for each platform. So the climate tests should include those tests.

Reply:
@MartinHjelmare new test added, did I understand what you were asking for?

Reply:
Yes, that looks good for the state check. Then for the services here we should also assert that the device methods have been called after the corresponding service call. For that we need access to the mocked device in this test.

Reply:
@MartinHjelmare done

Reply:
Nice! Maybe assert the arguments of the calls too, but looks good!

Reply:
Fixed as well as Im already warmed up. Thanks @MartinHjelmare !